docs/rest_api.md

```diff
@@ -0,0 +1,147 @@
+# REST API
+
+`TODO`: better doc.
+## Features Stores
+
+The plugin will store its components in an elasticsearch index:
+
+Initialize the default store:
+```
+PUT /_ltr
+```
+
+Initialize a custom store:
+```
+PUT /_ltr/custom
+```
+
+Stores can be deleted:
+
+```
+DELETE /_ltr
+```
+
+or a custom store:
+```
+DELETE /_ltr/custom
+```
+
+
+## Features
+
+`/_ltr/_feature/feature_name` supports `PUT`, `POST`, `GET` and `DELETE` operations.
+
+The format of a feature is :
+```json
+{
+  "name": "my_feature",
+  "params": ["query_string"],
+  "template_language": "mustache",
+  "template" : {
+    "match": {
+      "field": "{{query_string}}"
+    }
+  }
+}
+```
+
+To use a custom store simply prefix with the name of the store:
+`/_ltr/custom_store/_feature/feature_name`
+
+## Feature Sets
+
+`/_ltr/_featureset/featureset_name` supports `PUT`, `POST`, `GET` and `DELETE` operations.
+
+The format of a feature set is :
+```json
+{
+  "name": "my_feature_set",
+  "features" : [
+    {
+      "name": "my_feature",
+      "params": ["query_string"],
+      "template_language": "mustache",
+      "template" : {
+        "match": {
+          "field": "{{query_string}}"
+        }
+      }
+    }
+  ]
+}
+```
+
+Feature names in the set must be unique.
+
+To use a custom store simply prefix with the name of the store:
+`/_ltr/custom_store/_feature/feature_name`
+
+### Append features to a set
+A set can also be created/updated from existing features:
+
+`POST /_ltr/_featureset/my_featureset/_addfeatures/my_feature*`
```

Comment:
After discussion with @nomoa, this operation actually fails the second time as the features have already been added. POST make sense.

:+1: